kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaServiceReconciler.java

```diff
@@ -8,52 +8,200 @@
+ * actually exist, setting a {@link Condition.Type#ResolvedRefs} {@link Condition} accordingly.</p>
+ *
+ * <p>Because a service CR is not uniquely associated with a {@code KafkaProxy} it's not possible
+ * to set an {@code Accepted} condition on CR instances.</p>
+ */
 public final class KafkaServiceReconciler implements
         io.javaoperatorsdk.operator.api.reconciler.Reconciler<KafkaService> {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(KafkaServiceReconciler.class);
 
+    public static final String SECRETS_EVENT_SOURCE_NAME = "secrets";
+    public static final String CONFIG_MAPS_EVENT_SOURCE_NAME = "configmaps";
+
+    public static AnyLocalRef asRef(TrustAnchorRef reference) {
+        AnyLocalRef result = new AnyLocalRef();
+        result.setGroup(reference.getGroup());
+        result.setKind(reference.getKind());
+        result.setName(reference.getName());
+        return result;
+    }
+
     private final KafkaServiceStatusFactory statusFactory;
 
     public KafkaServiceReconciler(Clock clock) {
         this.statusFactory = new KafkaServiceStatusFactory(clock);
     }
 
+    @Override
+    public List<EventSource<?, KafkaService>> prepareEventSources(EventSourceContext<KafkaService> context) {
+        InformerEventSourceConfiguration<Secret> serviceToSecret = InformerEventSourceConfiguration.from(
+                Secret.class,
+                KafkaService.class)
+                .withName(SECRETS_EVENT_SOURCE_NAME)
+                .withPrimaryToSecondaryMapper((KafkaService cluster) -> ResourcesUtil.localRefAsResourceId(cluster, cluster.getSpec().getTls().getCertificateRef()))
+                .withSecondaryToPrimaryMapper(proxy -> ResourcesUtil.findReferrers(context,
+                        proxy,
+                        KafkaService.class,
+                        service -> service.getSpec().getTls().getCertificateRef()))
+                .build();
+        InformerEventSourceConfiguration<ConfigMap> serviceToConfigMap = InformerEventSourceConfiguration.from(
+                ConfigMap.class,
+                KafkaService.class)
+                .withName(CONFIG_MAPS_EVENT_SOURCE_NAME)
+                .withPrimaryToSecondaryMapper(
+                        (KafkaService cluster) -> ResourcesUtil.localRefAsResourceId(cluster, asRef(cluster.getSpec().getTls().getTrustAnchorRef())))
+                .withSecondaryToPrimaryMapper(proxy -> ResourcesUtil.findReferrers(context,
+                        proxy,
+                        KafkaService.class,
+                        service -> asRef(service.getSpec().getTls().getTrustAnchorRef())))
+                .build();
+        return List.of(
+                new InformerEventSource<>(serviceToSecret, context),
+                new InformerEventSource<>(serviceToConfigMap, context));
+    }
+
     @Override
     public UpdateControl<KafkaService> reconcile(KafkaService service, Context<KafkaService> context) {
```

Comment:
Note that we can't validate the protocols or cipher suites here because we don't know what's supported by the JVM of the operand. Instead we're going to have to pass them though unvalidated. It might be good if the proxy terminated in a well-defined way in this situation (ideally logging what _is_ supported), but it feels like it's a rough edge we'll have to live with.